Source/Particles/MultiParticleContainer.cpp

```diff
@@ -534,13 +535,57 @@ MultiParticleContainer::mapSpeciesProduct ()
         // pc->ionization_product_name and store its ID into
         // pc->ionization_product.
         if (pc->do_field_ionization){
-            int i_product = getSpeciesID(pc->ionization_product_name);
+            const int i_product = getSpeciesID(pc->ionization_product_name);
             AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
                 i != i_product,
                 "ERROR: ionization product cannot be the same species");
             pc->ionization_product = i_product;
         }
+
+#ifdef WARPX_QED
+        if (pc->has_breit_wheeler()){
+            const int i_product_ele = getSpeciesID(
+                pc->m_qed_breit_wheeler_ele_product_name);
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                i != i_product_ele,
+                "ERROR: Breit Wheeler product cannot be the same species");
+            pc->m_qed_breit_wheeler_ele_product = i_product_ele;
+
+            const int i_product_pos = getSpeciesID(
+                pc->m_qed_breit_wheeler_pos_product_name);
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                i != i_product_pos,
+                "ERROR: Breit Wheeler product cannot be the same species");
+            pc->m_qed_breit_wheeler_pos_product = i_product_pos;
+
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i_product_ele < allcontainers.size(),
+                "ERROR: Breit Wheeler product species not found");
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i_product_pos < allcontainers.size(),
+                "ERROR: Breit Wheeler product species not found");
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                allcontainers[i_product_ele]->AmIA<PhysicalSpecies::electron>() &&
+                allcontainers[i_product_pos]->AmIA<PhysicalSpecies::positron>(),
+                "ERROR: Breit Wheeler product species are of wrong type");
+        }
+
+        if(pc->has_quantum_sync()){
+            const int i_product_phot = getSpeciesID(
+                pc->m_qed_quantum_sync_phot_product_name);
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                i != i_product_phot,
+                "ERROR: Quantum Synchrotron product cannot be the same species");
+            pc->m_qed_quantum_sync_phot_product = i_product_phot;
+
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i_product_phot < allcontainers.size(),
+                "ERROR: Quantum Synchrotron product species not found");
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                allcontainers[i_product_phot]->AmIA<PhysicalSpecies::photon>(),
+                "ERROR: Quantum Synchrotron product species is of wrong type");
+        }
+#endif
```

Comment:
Does this piece of code really belong here? Maybe we could put it in a new function
```c++
#ifdef WARPX_QED
MultiParticleContainer::CheckQEDSpecies(){
    ...
}
#endif
```
that we would call after `mapSpeciesProduct`, what do you think?

Reply:
I have created a new `CheckQEDProductSpecies` function as suggested, which now includes all the checks (except the test on the existence of the product species). For consistency I have created an analogous function for the Ionization process. In the future we might decide to ask the user to mark as `electrons` the product species and check for that...